lib/internal/watch_mode/files_watcher.js

```diff
@@ -162,9 +162,18 @@ class FilesWatcher extends EventEmitter {
     if (this.#passthroughIPC) {
       this.#setupIPC(child);
     }
-    if (this.#mode !== 'filter') {
-      return;
-    }
+    /*
+    * TODO: Remove this comment
+    *
+    * I'm leaving this comment here to make sure we discuss this:
+    * Why were we filtering files only in filter mode?
+    *
+    * No tests are failing after removing this condition.
+    *
+    * if (this.#mode !== 'filter') {
+    *   return;
+    * }
+    */
```

Comment:
Which PR introduced this? Can you ping the original author?

Reply:
If I'm not mistaken, this change is from @MoLow

Reply:
I'm not sure what to make of this change 😄

Reply:
I think it would be great to know if that logic was there for a specific reason. Removing it didn't disrupt any tests.

I added the comment just to ensure that the change didn’t go unnoticed.

Reply:
Il need to get back from vacation to be in front of a computer to be sure of my answer, but using a single watcher (i.e all mode is probably more efficient and peformant) so creating additional watchers when they are not needed just adds more overhead that is unneeded

Reply:
I agree with this.